### ISSUE TITLE
Crash when reopening a QML/Web based panel that was closed on a different monitor

### ISSUE THREAD

**Issue by JosephCoppola-FW** [githubuser0xFFFF/Qt-Advanced-Docking-System#76]
This is pretty easy to reproduce, and I was able to reproduce it in the advanced demo.

In demo, add webenginewidgets to the demo.pro, and then import the header `#include <QWebEngineView>` in MainWindow.cpp

Then in createContent() add the following below:

```
    QWebEngineView* test = new QWebEngineView();
    test->setUrl(QUrl("https://www.google.com"));
    ads::CDockWidget* web = new ads::CDockWidget("Web");
    web->setWidget(test);
    ViewMenu->addAction(web->toggleViewAction());
    DockManager->addDockWidget(ads::CenterDockWidgetArea, web);
```

With this code added, now follow these steps:

Steps to Repro:
1. Move Main Window to a 2nd monitor.
2. Undock the Web panel.
3. Close the Web panel via the windows close button.
4. Move Main Window to your 1st monitor.
5. Attempt to re-open the Web panel via the view menu.

In debug, you will see the below. I was able to reproduce this in QT 5.12.3 and 5.13.0. Please note that if the floating panel is closed via the action in the view menu in place of step 3 above, the crash does not occur.

![image](https://user-images.githubusercontent.com/37883740/67975454-1c732680-fbe2-11e9-9b23-1711f8bff3cf.png)

**Comment by JosephCoppola-FW** [comment]
So I think I found a work around. 

The crash seems related to the closeEvent in FloatingDockContainer.cpp.

Rather than calling `Super::closeEvent(event)`, replacing that call with the code below seems to address the issue. 

```
event->ignore()
hide()
```

This addresses the crash I am seeing, but I'm not sure of the full implications of ignoring the close event and simply hiding instead. I'm going to do some extensive testing to see the ramifications of this.

**Comment by JosephCoppola-FW** [comment]
First off, thanks for commenting your code so efficiently. 

Because of that, I was able to find your ticket here https://bugreports.qt.io/browse/QTBUG-74649 and saw that this specific bug was fixed in 5.12.3, the version we happen to be using. 

So I was able to use the solution above, where I ignore close and just hide, exactly the same as you do for versions 5.9.2 to 5.12.2. 

Then in the event function in FloatingDockContainer.cpp I updated the NonClientAreaMouseButtonPress check to include `&& QGuiApplication::mouseButtons().testFlag(Qt::LeftButton)` again and docking works as expected again.
